### PR TITLE
v1.4: Command/select-ingredient

### PR DESCRIPTION
**Pull request by rebstan97**

### CODE REVIEW

File: src/main/java/seedu/restaurant/logic/commands/ingredient/SelectIngredientCommand.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        EventsCenter.getInstance().post(new JumpToIngredientListRequestEvent(targetIndex));
+        return new CommandResult(String.format(MESSAGE_SELECT_INGREDIENT_SUCCESS, targetIndex.getOneBased()));
+
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof SelectIngredientCommand // instanceof handles nulls
+                && targetIndex.equals(((SelectIngredientCommand) other).targetIndex)); // state check
```
Comment (AZhiKai) [inline src/main/java/seedu/restaurant/logic/commands/ingredient/SelectIngredientCommand.java]: Indent `&&`

File: src/test/java/seedu/restaurant/logic/commands/ingredient/SelectIngredientCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
+package seedu.restaurant.logic.commands.ingredient;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static seedu.restaurant.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.restaurant.logic.commands.CommandTestUtil.showIngredientAtIndex;
+import static seedu.restaurant.logic.commands.menu.MenuCommandTestUtil.assertCommandFailure;
```
Comment (HyperionNKJ) [inline src/test/java/seedu/restaurant/logic/commands/ingredient/SelectIngredientCommandTest.java]: I suggest creating an IngredientCommandTestUtil and implementing your own assertCommandFailure.

File: src/test/java/guitests/guihandles/menu/IngredientStackPanelHandle.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
+package guitests.guihandles.menu;
```
Comment (HyperionNKJ) [inline src/test/java/guitests/guihandles/menu/IngredientStackPanelHandle.java]: This class should be placed in the ingredient package.

File: src/test/java/seedu/restaurant/logic/commands/ingredient/SelectIngredientCommandTest.java (hunk 2 above)
Comment (AZhiKai) [reply src/test/java/seedu/restaurant/logic/commands/ingredient/SelectIngredientCommandTest.java]: The thing here is it seems like the method is doing the same thing. If so, we should extract it out.

File: src/test/java/guitests/guihandles/menu/IngredientStackPanelHandle.java (hunk 3 above)
Comment (rebstan97) [reply src/test/java/guitests/guihandles/menu/IngredientStackPanelHandle.java]: Oh my, thanks for pointing out

**Pull request merged**